### DISCUSSION TITLE
How to catch a message deleted by a user

### DISCUSSION THREAD

**Discussion by NaPs** [python-telegram-bot/python-telegram-bot#3178] [Q&A] [answered]
Hi folks, I'm trying to make my bot able to react to messages and attachments deleted by users. I cannot find any handler or filter to do that in doc but it seems that clients are able to do that since they can display the "message deleted" notice. How to do that with PTB?

**Comment by Poolitzer** [comment, marked as answer]
That's impossible, bots dont get an update when messages get deleted.

Just because the client api can do something, doesn't sadly mean bots can

**Comment by Khidr01** [comment]
Why you do not export all the written chats to a text file, or find some other way like this to keep getting all the messages? Therefore, even if the user deleted the message, you still have the whole story.
